File: go/two_sum.go
package main

func twoSum(nums []int, target int) []int {
	values := make(map[int]int)
	for index, value := range nums {
		if number, exists := values[target-value]; exists {
			return []int{number, index}
		}
		values[value] = index
	}
	return nums
}

File: python/two_sum.py
from typing import List, Dict

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        number_index: Dict[int, int] = {}
        for index, number in enumerate(nums):
            remainder = target - number
            
            if remainder in number_index:
                return [number_index[remainder], index]
            number_index[number] = index

        return nums